scripts/apps/search/components/fields/authors.tsx

```diff
@@ -0,0 +1,166 @@
+/* eslint-disable react/no-multi-comp */
+
+import React from 'react';
+
+import {IUser, IRestApiResponse, IVocabulary, IVocabularyItem, IAuthorsFieldOptions} from 'superdesk-api';
+import {IPropsItemListInfo} from '../ListItemInfo';
+import {SuperdeskReactComponent} from 'core/SuperdeskReactComponent';
+import {getVocabularyItemNameTranslated, gettext} from 'core/utils';
+import {Popover} from 'superdesk-ui-framework/react';
+
+interface IProps {
+    authors: Array<{userId: IUser['_id'], roleId: string}>;
+    options: IAuthorsFieldOptions;
+}
+
+interface IState {
+    authors?: Array<{user: IUser, roleQcode: string}>;
+    vocabularyItems?: Map<string, IVocabularyItem>; // indexed by qcode
+}
+
+const SEPARATOR = <span style={{opacity: 0.5}}> / </span>;
+const AUTHORS_TO_SHOW_AT_ONCE: number = 2;
+
+export class AuthorsComponent extends SuperdeskReactComponent<IProps, IState> {
+    constructor(props: IProps) {
+        super(props);
+
+        this.state = {};
+    }
+    componentDidMount() {
+        const {authors} = this.props;
+
+        const rolesByUserId = authors.reduce((acc, item) => {
+            acc[item.userId] = item.roleId;
+
+            return acc;
+        }, {});
+
+        const userIds = authors.map(({userId}) => userId);
+
+        const getUsers = () => this.asyncHelpers.httpRequestJsonLocal<IRestApiResponse<IUser>>({
```

Comment:
I think we should handle this in some layer and not send a query for every item in the list, we could debounce that query but keep all the requested ids and then fetch it just once.

Reply:
I have added such a layer on a separate branch and will submit a PR after this is merged.